src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoMockStaticToMockito.java

```diff
@@ -327,154 +297,240 @@ private static List<Expression> getMockedTypesFromPrepareForTestAnnotation(List<
-            return methodBody.getStatements().stream().filter(statement -> statement instanceof J.MethodInvocation)
-              .map(J.MethodInvocation.class::cast)
-              .filter(MOCKED_STATIC_MATCHER::matches)
-              .filter(methodInvocation -> methodInvocation.getSelect() instanceof J.Identifier)
-              .anyMatch(methodInvocation -> ((J.Identifier) methodInvocation.getSelect()).getSimpleName()
-                .equals(staticMock.getSimpleName()));
+        @NotNull
+        private J.ClassDeclaration addFieldDeclarationForMockedTypes(J.ClassDeclaration classDecl, ExecutionContext ctx, List<Expression> mockedTypes) {
+            // Get the actually invoked staticMethod by class
+            Map<String, J.MethodInvocation> invocationByClassName = getCursor().getNearestMessage(MOCK_STATIC_INVOCATIONS);
+            if (invocationByClassName == null || invocationByClassName.isEmpty()) {
+                // If there are no invocations, nothing needs to be done here
+                return classDecl;
+            }
+            // Add field declarations of mocked types
+            Map<J.Identifier, Expression> mockedTypesIdentifiers = new LinkedHashMap<>();
+            for (Expression mockedType : mockedTypes) {
+                JavaType.Parameterized classType = TypeUtils.asParameterized(mockedType.getType());
+                if (classType == null) {
+                    continue;
+                }
+                JavaType.FullyQualified fullyQualifiedMockedType = TypeUtils.asFullyQualified(classType.getTypeParameters().get(0));
+                if (fullyQualifiedMockedType == null) {
+                    continue;
+                }
+                String classlessTypeName = fullyQualifiedMockedType.getClassName();
+                if (invocationByClassName.get(classlessTypeName + ".class") == null) {
+                    // Only add fields for classes that are actually invoked by mockStatic()
+                    // The not mocked class can be removed from import
+                    maybeRemoveImport(fullyQualifiedMockedType.getFullyQualifiedName());
+                    continue;
+                }
+                String mockedTypedFieldName = MOCK_PREFIX + classlessTypeName;
+                if (isFieldAlreadyDefined(classDecl.getBody(), mockedTypedFieldName)) {
+                    continue;
+                }
+                classDecl = classDecl.withBody(classDecl.getBody()
+                  .withTemplate(
+                    JavaTemplate.builder(() -> getCursor().getParentTreeCursor(),
+                        "private MockedStatic<#{}> " + MOCK_PREFIX + "#{};")
+                      .javaParser(() -> JavaParser.fromJavaVersion()
+                        .classpathFromResources(ctx, "mockito-core-3.12.4")
+                        .build())
+                      .staticImports("org.mockito.Mockito.mockStatic")
+                      .imports(MOCKED_STATIC)
+                      .build(),
```

Comment:
This may be missing a calls `JavaTemplate.Builder.import()` for `MockedStatic` and `fullyQualifiedType`. Possible source of missing type attribution as surfaced in the test cases.